Content.Server/Chat/ChatConditions/AdminOOCEnabledSessionChatCondition.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Linq;
+using Content.Server.Station.Components;
+using Content.Server.Station.Systems;
+using Content.Shared.CCVar;
+using Content.Shared.Chat;
+using Content.Shared.Ghost;
+using Robust.Shared.Configuration;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+
+namespace Content.Server.Chat.ChatConditions;
+
+[DataDefinition]
+public sealed partial class AdminOOCEnabledSessionChatCondition : SessionChatCondition
+{
+    [Dependency] private readonly IConfigurationManager _configurationManager = default!;
+
+    public override HashSet<ICommonSession> FilterConsumers(HashSet<ICommonSession> consumers, Dictionary<Enum, object> channelParameters)
+    {
+        IoCManager.InjectDependencies(this);
+
+        return _configurationManager.GetCVar<bool>(CCVars.AdminOocEnabled) ? consumers : new HashSet<ICommonSession>();
```

Comment:
How does this returning hash set is used? if other code is not adding stuff right into it - it might be better to use some reference to immutable empty entity - then u won't have to re-created hashsets, polluting memory with useless stuff.